[BUG] Pascal has gone missing!

We just spotted that Pascal has vanished from the live site!

Probably this commit https://github.com/mattgodbolt/compiler-explorer/commit/e52f66017c7972d1ab3b507e545aee6191a31b52

Resulting in this error at startup
`warn: Unable to stat fpc304 compiler binary: Cannot find module '../demangler-pascal' {"code":"MODULE_NOT_FOUND","stack":"Error: Cannot find module '../demangler-pascal'\n    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)\n    at Function.Module._load (internal/modules/cjs/loader.js:508:25)\n    at Module.require (internal/modules/cjs/loader.js:637:17)\n    at require (internal/modules/cjs/helpers.js:22:18)\n    at new BaseCompiler (/opt/compiler-explorer/ce/lib/base-compiler.js:72:35)\n    at new FPCCompiler (/opt/compiler-explorer/ce/lib/compilers/pascal.js:35:9)\n    at CompileHandler.create (/opt/compiler-explorer/ce/lib/handlers/compile.js:110:37)"}
`

--- a/etc/config/pascal.defaults.properties
+++ b/etc/config/pascal.defaults.properties
@@ -8,7 +8,7 @@ options=
 versionFlag=-h | grep "Compiler version" 
 supportsBinary=true
 compilerType=pascal
-demanglerClassFile=../demangler-pascal
+demanglerClassFile=./demangler-pascal
 
 #################################
 #################################

--- a/lib/compilers/pascal.js
+++ b/lib/compilers/pascal.js
@@ -34,13 +34,13 @@ class FPCCompiler extends BaseCompiler {
     constructor(info, env) {
         super(info, env);
 
-        let demanglerClassFile = this.compiler.demanglerClassFile;
-        if (!demanglerClassFile) demanglerClassFile = "../demangler-pascal";
-        const demanglerClass = require(demanglerClassFile).Demangler;
-
-        this.demangler = new demanglerClass(null, null, this);
         this.compileFilename = 'output.pas';
         this.supportsOptOutput = false;
+    }
+
+    processAsm(result, filters) {
+        this.demangler = new this.demanglerClass(null, this);
+        return this.asm.process(result.asm, filters);
     }
 
     postProcessAsm(result, filters) {

--- a/lib/demangler-pascal.js
+++ b/lib/demangler-pascal.js
@@ -27,8 +27,8 @@ const SymbolStore = require("./symbol-store").SymbolStore;
 const Demangler = require("./demangler").Demangler;
 
 class DemanglerPascal extends Demangler {
-    constructor(demanglerExe, symbolstore) {
-        super(demanglerExe, symbolstore);
+    constructor(demanglerExe, compiler) {
+        super(demanglerExe, compiler);
 
         this.symbolStore = new SymbolStore();
         this.fixedsymbols = {};

--- a/test/pascal-tests.js
+++ b/test/pascal-tests.js
@@ -311,6 +311,8 @@ describe('Pascal ASM line number injection', function () {
     };
 
     const compiler = new PascalCompiler(info, ce);
+    compiler.demanglerClass = require("../lib/demangler-pascal").Demangler;
+    compiler.demangler = new compiler.demanglerClass(null, compiler);
 
     it('Should have line numbering', function () {
         return new Promise(function (resolve, reject) {

